src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/ReadOperation.java

```diff
@@ -75,28 +82,26 @@ default Uni<FindResponse> findDocument(
         .onItem()
         .transform(
             rSet -> {
-              int remaining = rSet.getRowsCount();
-              int colCount = rSet.getColumnsCount();
+              int remaining = rSet.remaining();
               List<ReadDocument> documents = new ArrayList<>(remaining);
-              Iterator<QueryOuterClass.Row> rowIterator = rSet.getRowsList().stream().iterator();
+              Iterator<Row> rowIterator = rSet.currentPage().iterator();
               while (--remaining >= 0 && rowIterator.hasNext()) {
-                QueryOuterClass.Row row = rowIterator.next();
+                Row row = rowIterator.next();
                 ReadDocument document = null;
                 try {
-                  JsonNode root =
-                      readDocument ? objectMapper.readTree(Values.string(row.getValues(2))) : null;
+                  JsonNode root = readDocument ? objectMapper.readTree(row.getString(2)) : null;
```

Comment:
we are relying on positional values? this seems brittle.

Reply:
Since we use Stargate api query builder this is the only way to bind the values.